pkg/puller/puller.go

```diff
@@ -330,34 +289,28 @@ func (p *Puller) syncPeer(ctx context.Context, peer swarm.Address, po, d uint8)
 
 	// peer outside depth?
 	if po < d && po > 0 {
-		cur, bin := c[po], po
-		// start just one bin for historical and live
-		binCtx, cancel := context.WithCancel(ctx)
-		syncCtx.binCancelFuncs[po] = cancel
-		if cur > 0 {
-			p.wg.Add(1)
-			go p.histSyncWorker(binCtx, peer, bin, cur) // start historical
-		}
-		p.wg.Add(1)
-		go p.liveSyncWorker(binCtx, peer, bin, cur) // start live
-
+		p.syncPeerBin(ctx, syncCtx, peer, po, c[po])
```

Comment:
do not sync outside NN pretty please. this could be a (debug) API call used when. a node wants to quickly fill a. lot of storage beside what is currently. its neighbourhood.

Reply:
i split this to a separate issue (#818) because this PR already changes some things in pull sync and this would be a big changeset to test in one go. i'd be happy to have it in a separate pr, with separate testing, to see that there are no regressions